src/ParserParameterProcessor.php

```diff
@@ -140,15 +140,20 @@ public function setParameters( array $parameters ) {
 	}
 
 	/**
-	 * Add parameter key and value
-	 *
 	 * @since 1.9
 	 *
 	 * @param string $key
-	 * @param string $value
+	 * @param string|array $value
 	 */
 	public function addParameter( $key, $value ) {
-		if( $key !== '' && $value !== '' ) {
+
+		if ( $key === '' ) {
+			return null;
+		}
+
+		if ( is_array( $value ) ) {
+			$this->parameters[$key] = $value;
```

Comment:
> In this case you replace any existing values while for the else part you merge the thing in. Is that intended?

It is kind of blurry because it could be seen as a `set` instead of an `add` but I wanted to avoid to add a new method to avoid:

```
foreach ( $values as $value ) {
  $parserParameterProcessor->addParameter( 'foo', $value )
}
```

```
$parserParameterProcessor->addParameter( 'foo', $values )
```